healthyregions/SDOHPlace-MetadataManager
# feat: added Docker recipe for building + running the MetadataManager

## Problem
For consistency in deployment, we could create a Docker image for the MetadataManager

## Approach
* feat: adjusted Dockerfile to build + run the application on port 8000
* feat: added a Docker Compose file to simplify the commands needed to run Solr along with the MetadataManager
    * `docker compose up -d --build` can be used to build + run the app in one step

## How to Test
1. Checkout and run this branch locally
2. Build the image: `docker compose build`
    * NOTE: this is a shorthand for running `docker build -t herop/sdoh-metadata-manager .`
3. Run the image: `docker compose up -d`
    * NOTE: this is a shorthand for running `docker run -itd -p 8000:8000 --name sdoh-manager herop/sdoh-metadata-manager`
    * The first time that the application runs, Solr will pre-create a new core named `blacklight-core-dev`
    * You can view the Solr web UI by navigating your browser to http://localhost:8983
    * Once Solr starts up, the manager app will start as well
4. Register the index data from Solr: `docker exec -it sdoh-manager flask registry index --clean`
5. Wait a few minutes for the application to complete its initial data load (this can take awhile, since it loads ~102 records one-by-one)
    * You can view the solr/manager logs with `docker compose logs -f`
    * You can view the [Solr Core](http://localhost:8983/solr/#/blacklight-core-dev/core-overview) in your browser to see the progress using `Num Docs` at the top-left. Note that this page does not refresh automatically in realtime.
6. Once the data load is complete, you should see that the Flask app is served on port 8000
7. Navigate your browser to http://localhost:8000 to view the SDOH Metadata Manager with the loaded data
    * This port can be easily mapped/overridden at runtime
8. In SDOHPlace NextJS Discovery App, open up `.env` and set the following config:
```env
NEXT_PUBLIC_SOLR_URL='http://localhost:8983/solr/blacklight-core-dev'
```
9. Start up the SDOHPlace Discovery App and navigate to http://localhost:3000/search
    * You should see on the left side that results are being loaded from Solr 
10. Shutdown the containers: `docker compose down`
    * NOTE: this is a shorthand for running `docker rm -f sdoh-manager`
